File: test/end-to-end/Graphs/Ruby/data/weighted_node.rb

##
# A weighted, directed node in a graph.
##
class WeightedNode
    attr_accessor :data
    attr_accessor :edges

    def initialize(data)
        self.data = data
        self.edges = []
    end

    def get_neighbors_in_order()
        nodes = []

        for edge in self.edges
            nodes.push(edge.to)
        end

        self.edges.sort_by! {|edge| edge.distance}

        return nodes
    end
end

File: test/end-to-end/Graphs/PHP/data/weighted_node.php
<?php

/**
 * A weighted, directed node in a graph.
 */
class WeightedNode {
    public $id;
    public $data;
    private $edges;

    function __construct($id, $data) {
        $this->id = $id;
        $this->data = $data;
        $this->edges = [];
    }

    function getNeighborsInOrder() {
        $nodes = [];

        foreach ($this->edges as $edge) {
            array_push($nodes, $edge->to);
        }

        usort($this->edges, function($edgeA, $edgeB) { return $edgeA->distance - $edgeB->distance; });

        return $nodes;
    }
}
